src/Command.ts

```diff
@@ -26,10 +30,21 @@ export interface Command {
   readonly options?: ApplicationCommandOptionData[];
 
   /**
-   * The function that gets executed after the command
-   * is invoked.
+   * The function that gets executed after the command is invoked.
+   * @param args
+   * @param args.interaction Interaction object from discord.js
+   * @param args.registerUI **Must be called at most once per message!**
+   * Generates a discord.js compatible UI from Dispatch components.
    */
-  run(interaction: CommandInteraction): Promise<void> | void;
+  run({
+    interaction,
+    registerUI,
+  }: {
+    interaction: CommandInteraction;
+    registerUI: (
+      ui: UIComponent | UIComponent[] | UIComponent[][]
+    ) => MessageActionRow[];
+  }): Promise<void> | void;
```

Comment:
I don't know how I feel about `registerUI`being a parameter on `run`. registerUI doesn't seem like something that should be related to running the command, especially, if you don't use components.

Reply:
That's why I used the destructuring syntax so you don't need to put it in your parameter list unless you need it. This will also make it easy to inject other dependencies like storage later without introducing more parameters or a IOC container.

Reply:
But why does this need to be passed in as a function parameter in the first place?

Reply:
It doesn't, I just thought a function injection was simpler than passing in the entire `Client` since you really only need the function.

Reply:
I feel like having a class-based command with a client constructor (like other frameworks) would be easier to reason about, it would also allow for adding other properties later.